components/gitpod-db/src/typeorm/project-db-impl.ts

```diff
@@ -23,12 +23,16 @@ function toProjectEnvVar(envVarWithValue: ProjectEnvVarWithValue): ProjectEnvVar
 }
 
 @injectable()
-export class ProjectDBImpl implements ProjectDB {
-    @inject(TypeORM) typeORM: TypeORM;
-    @inject(EncryptionService) protected readonly encryptionService: EncryptionService;
+export class ProjectDBImpl extends TransactionalDBImpl<ProjectDB> implements ProjectDB {
+    constructor(
+        @inject(EncryptionService) protected readonly encryptionService: EncryptionService,
```

Comment:
❤️ Love this! Would be great to use constructor injection everywhere from now on.
Also, let's make such fields `private` rather than `protected`.